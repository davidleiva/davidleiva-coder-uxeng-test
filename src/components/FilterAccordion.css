.MuiAccordion-root.MuiPaper-root {
    background-color: rgb(4, 6, 11);
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    box-shadow: none;
    background-image: none;
    color: hsl(218, 32%, 77%);
  }
  
  .MuiAccordion-root.MuiPaper-root svg path {
    fill: hsl(218, 32%, 77%);
  }
  
  .MuiAccordion-root.MuiPaper-root:hover,
  .MuiAccordion-root.MuiPaper-root.Mui-expanded {
    color: hsl(220, 55%, 98%);
  }
  
  .MuiAccordion-root.MuiPaper-root:hover svg path {
    color: hsl(220, 55%, 98%);
  }
  
  .MuiAccordionDetails-root {
    padding: 0;
  }
  
  .MuiAccordion-root.MuiPaper-root .MuiAccordionSummary-root {
    padding-left: 0;
    padding-right: 0;
    height: 36px;
  }
  
  .MuiAccordionSummary-content > .MuiTypography-root,
  .MuiListItemText-root > .MuiTypography-root,
  .FilterSideBar .MuiAccordionSummary-content > .MuiTypography-root,
  .MuiAccordionSummary-content > .MuiTypography-root,
  .FilterSideBar .MuiListItemText-root > .MuiTypography-root {
    font-size: 12px;
  }
  
  .MuiAccordion-root.MuiPaper-root:before {
    content: none;
  }
  
  .MuiAccordion-root.MuiPaper-root .MuiSvgIcon-root {
    width: 0.66em;
    height: 0.66em;
  }
  
  .MuiAccordionSummary-content.Mui-expanded {
    margin: 4px 0;
  }
  
  .MuiAccordion-root .MuiButtonBase-root {
    min-height: initial;
    height: 36px;
  }
  
  .MuiAccordion-root.MuiPaper-root.Mui-expanded {
    margin-bottom: 0;
    margin-top: 0;
  }
  
  .MuiListItemButton-root.justifyEnd .MuiListItemIcon-root {
    justify-content: end;
  }

  .FilterBar__Divider.MuiDivider-root {
    margin: 8px auto;
  }

.FilterSearchUser input, .FilterSearchUser label,
.FilterSearchUser .MuiAvatar-circular {
  font-size: 12px;
}

.MuiListItemButton-root .MuiListItemIcon-root, .MuiListItemButton-root .MuiAvatar-root {
  min-width: 0;
}

.MuiListItemButton-root .MuiAvatar-root {
  margin-right: 8px;
}