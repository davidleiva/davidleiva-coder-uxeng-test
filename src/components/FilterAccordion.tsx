import Accordion from '@mui/material/Accordion';
import AccordionSummary from '@mui/material/AccordionSummary';
import AccordionDetails from '@mui/material/AccordionDetails';
import Typography from '@mui/material/Typography';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { FiltersSB } from './FiltersSB';

export interface FilterAccordionProps {
    title: string,
    defaultExp?: boolean,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    filters: any,
    isExpanded: boolean,
    // eslint-disable-next-line @typescript-eslint/ban-types
    handleChange: Function,
    filterState?: object,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    type: any
}

export const FilterAccordion = ({
    title,
    defaultExp,
    filters,
    isExpanded,
    handleChange,
    type
    }: FilterAccordionProps) => {
    
    return (
        <Accordion
            expanded={isExpanded}
            sx={{
                bgcolor: 'transparent',
            }}
            defaultExpanded={defaultExp}
            onChange={() => handleChange()}
        >
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
                aria-controls="panel1a-content"
                id="panel1a-header"
            >
                <Typography>{title}</Typography>
            </AccordionSummary>
            <AccordionDetails>
                <FiltersSB filters={filters} type={type}/>
            </AccordionDetails>
        </Accordion>
      )
};

export default FilterAccordion;