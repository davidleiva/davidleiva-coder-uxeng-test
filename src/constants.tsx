export const FILTER_TYPES = [
    'predefined','resource_type', 'action','usernames','build_reason',
];

export const RESOURCES = [
    'api_key', 'convert_login', 'git_ssh-key', 'convert_login', 'group', 'license', 'template', 'template_version', 'user', 'workspace', 'workspace_build'
];

export const ACTIONS = [
    'create', 'delete', 'login', 'logout', 'register', 'start', 'stop', 'write' 
];

export const USERS = [
    { name: 'David Leiva' },
    { name: 'Bruno Quaresma'}
];

export const PREDEFINED_FILTERS = [
    {
        label: 'Created workspaces',
        value: { resource_type: RESOURCES[9], action: ACTIONS[0] },
    },
    {
        label: 'Added templates',
        value: { resource_type: RESOURCES[6], action: ACTIONS[0] },
    },
    {
        label: 'Deleted users',
        value: { resource_type: RESOURCES[8], action: ACTIONS[1] },
    },
    {
        label: 'Builds started by a user',
        value: { resource_type: RESOURCES[10], action: ACTIONS[5], build_reason: 'initiator' },
    },
    {
        label: 'User logins',
        value: { resource_type: RESOURCES[0], action: ACTIONS[2] },
    }
];

export const RESOURCE_FILTERS = [
    {
        label: 'Api Key',
        value: { resource_type: RESOURCES[0] },
    },
    {
        label: 'Login',
        queryValue: RESOURCES[1],
        value: { resource_type: RESOURCES[1] },
    },
    {
        label: 'Git SSH Key',
        value: { resource_type: RESOURCES[2] },
    },
    {
        label: 'Group',
        value: { resource_type: RESOURCES[3] },
    },
    {
        label: 'Template',
        value: { resource_type: RESOURCES[4] },
    },
    {
        label: 'Template version',
        value: { resource_type: RESOURCES[5] },
    },
    {
        label: 'User',
        value: { resource_type: RESOURCES[6] },
    },
    {
        label: 'Workspace',
        value: { resource_type: RESOURCES[7] },
    },
    {
        label: 'Workspace Build',
        value: { resource_type: RESOURCES[8] },
    }
];

// export const RESOURCE_FILTERS = [
//     {
//         label: 'Api Key',
//         value: { action: ACTIONS[0] },
//     },
//     {
//         label: 'Login',
//         value: { action: ACTIONS[1] },
//     },
//     {
//         label: 'Git SSH Key',
//         value: { action: ACTIONS[2] },
//     },
//     {
//         label: 'Group',
//         value: { action: ACTIONS[3] },
//     },
//     {
//         label: 'Template',
//         value: { action: ACTIONS[4] },
//     },
//     {
//         label: 'Template version',
//         value: { action: ACTIONS[5] },
//     },
//     {
//         label: 'User',
//         value: { action: ACTIONS[6] },
//     },
//     {
//         label: 'Workspace',
//         value: { action: ACTIONS[7] },
//     },
//     {
//         label: 'Workspace Build',
//         value: { action: ACTIONS[8] },
//     }
// ];

export const ACTION_FILTERS = [
    {
        label: 'Api Key',
        value: { action: ACTIONS[0] },
    },
    {
        label: 'Login',
        value: { action: ACTIONS[1] },
    },
    {
        label: 'Git SSH Key',
        value: { action: ACTIONS[2] },
    },
    {
        label: 'Group',
        value: { action: ACTIONS[3] },
    },
    {
        label: 'Template',
        value: { action: ACTIONS[4] },
    },
    {
        label: 'Template version',
        value: { action: ACTIONS[5] },
    },
    {
        label: 'User',
        value: { action: ACTIONS[6] },
    },
    {
        label: 'Workspace',
        value: { action: ACTIONS[7] },
    },
    {
        label: 'Workspace Build',
        value: { action: ACTIONS[8] },
    }
];

