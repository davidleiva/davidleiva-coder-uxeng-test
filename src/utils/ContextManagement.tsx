import { createContext } from "react";

export interface FilterState {
    resource_type: string,
    action: string,
    build_reason: string
}
export interface FilterStatesContextData {
    filters: FilterState;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    setFiltersState: (filter:any) => void;
}

// eslint-disable-next-line react-refresh/only-export-components
export const filtersContextDefaultValue: FilterStatesContextData = {
    filters: {resource_type: '', action: '', build_reason: ''},
    setFiltersState: () => null,
}

export const FilterStatesContext = createContext<FilterStatesContextData>(filtersContextDefaultValue);