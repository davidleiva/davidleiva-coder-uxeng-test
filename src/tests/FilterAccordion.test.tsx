import {describe, expect, test} from 'vitest';
import FilterAccordion from '../components/FilterAccordion';
import { PREDEFINED_FILTERS } from '../utils/constants';

describe("Accordion basic test", () => {
    test("FilterAccordion renders ok", () => {
        
        const fc = 
            <FilterAccordion 
                title={'title'}
                defaultExp={true}
                filters={PREDEFINED_FILTERS}
                isExpanded={true}
                handleChange={() => {}}
            />
        ;

        expect(fc).toMatchSnapshot()
    })
})