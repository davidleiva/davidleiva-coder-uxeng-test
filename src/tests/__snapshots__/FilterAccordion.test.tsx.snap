// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion basic test > FilterAccordion renders ok 1`] = `
<FilterAccordion
  defaultExp={true}
  filters={
    [
      {
        "label": "Created workspaces",
        "queryValue": "created_workspaces",
        "value": {
          "action": "create",
          "resource_type": "workspace",
        },
      },
      {
        "label": "Added templates",
        "queryValue": "added_templates",
        "value": {
          "action": "create",
          "resource_type": "template_version",
        },
      },
      {
        "label": "Deleted users",
        "queryValue": "deleted_users",
        "value": {
          "action": "delete",
          "resource_type": "user",
        },
      },
      {
        "label": "Builds started by a user",
        "queryValue": "builds_started",
        "value": {
          "action": "start",
          "build_reason": "initiator",
          "resource_type": "workspace_build",
        },
      },
      {
        "label": "User logins",
        "queryValue": "user_logins",
        "value": {
          "action": "login",
          "resource_type": "api_key",
        },
      },
    ]
  }
  handleChange={[Function]}
  isExpanded={true}
  title="title"
/>
`;
